### PR TITLE
ARROW-4291: [Dev] Support selecting features in release verification scripts

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by BryanCutler** [comment]
+1 also it would be nice to be selectively run integration tests. Right now it requires javascript to be also be built.

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -361,13 +362,35 @@ if [ "$ARTIFACT" == "source" ]; then
   TARBALL=apache-arrow-$1.tar.gz
   DIST_NAME="apache-arrow-${VERSION}"
 
+  # By default test all functionalities.
+  # To deactivate one test, deactivate the test and all of its dependents
+  TEST_JAVA=${TEST_JAVA:=1}
```
[the remark this answers is not in the document]
Comment (xhochy) [reply dev/release/verify-release-candidate.sh]: The `:-1` version was not compatible with the old bash version I had on OSX. I'll use the `: ${..}` path.

File: dev/release/verify-release-candidate.sh [hunk 2]
```diff
@@ -361,13 +362,35 @@ if [ "$ARTIFACT" == "source" ]; then
...
-  setup_miniconda
-  test_and_install_cpp
+  if [ ${TEST_JAVA} -gt 0 ]; then
+    test_package_java
+  fi
+  if [ ${TEST_CPP} -gt 0 ]; then
+    setup_miniconda
+    test_and_install_cpp
+  fi
   test_python
```
[the remark this answers is not in the document]
Comment (xhochy) [reply dev/release/verify-release-candidate.sh]: Yes, I've forgot those.

**Review by kou — APPROVED** [review]
+1

File: dev/release/verify-release-candidate.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (kou) [reply dev/release/verify-release-candidate.sh]: Oh, I didn't know that `${...:-...}` doesn't work with old bash.

**Pull request closed without merging**